engines/wasmtime/wasmtime.go

```diff
@@ -75,10 +77,20 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+// WithEngine allows you to override the default engine. Defaults to
+// `wasmtime.NewEngine`.
+func WithEngine(engine *wasmtime.Engine) *wasmtime.Engine {
+	return engine
+}
 
-func Engine() wapc.Engine {
-	return &engineInstance
+func Engine(engs ...*wasmtime.Engine) wapc.Engine {
```

Comment:
I think you meant to define a func type:

```suggestion
// EngineOption is a wasmtime-specific configuration, applied via Engine
type Option func(*engine)

func Engine(options ...EngineOption) wapc.Engine {
```

Reply:
see https://github.com/wapc/wapc-go/pull/40#discussion_r985022623